modules/broker-core/src/main/java/org/wso2/broker/core/store/SharedMessageStore.java

```diff
@@ -120,4 +120,12 @@ public void flush(long internalMessageId) {
     public Collection<Message> readStoredMessages(String queueName) throws BrokerException {
         return messageDao.readAll(queueName);
     }
+
+    /**
+     * Method to flush all pending messages.
+     */
+    public void flushAll() {
```

Comment:
Shall we change the method name to `clearPendingMessages()`? Since we are clearing the list